.github/workflows/cherry-pick.yml

```
@@ -1,26 +1,328 @@
+                    changelogEntry = false;
+                  }
+                } );
+
+                if ( changelogEntry === false ) {
+                  continue;
+                }
+
+                fs.readFile( './plugins/woocommerce/readme.txt', 'utf-8', function( err, data ) {
+                  if ( err ) {
+                    console.error( err );
+                  }
+
+                  changelogTxt = data.split( "\n" );
+                  let isInRange = false;
+                  let newChangelogTxt = [];
+
+                  for ( const line of changelogTxt ) {
+                    if ( isInRange === false && line === '== Changelog ==' ) {
+                      isInRange = true;
+                    }
+
+                    if ( isInRange === true && line.match( /\*\*WooCommerce Blocks/ ) ) {
+                      isInRange = false;
+                    }
+
+                    // Find the first match of the entry "Type".
+                    if ( isInRange && line.match( `\\* ${changelogEntryType} -` ) ) {
+                      newChangelogTxt.push( '* ' + changelogEntryType + ' - ' + changelogEntry + ` [#${{ needs.prep.outputs.pr }}](https://github.com/woocommerce/woocommerce/pull/${{ needs.prep.outputs.pr }})` );
+                      newChangelogTxt.push( line );
+                      isInRange = false;
+                      continue;
+                    }
+
+                    newChangelogTxt.push( line );
+                  }
+
+                  fs.writeFile( './plugins/woocommerce/readme.txt', newChangelogTxt.join( "\n" ), err => {
+                    if ( err ) {
+                      console.error( `Unable to generate the changelog entry for PR ${{ needs.prep.outputs.pr }}` );
+                    }
+                  } );
+                } );
+              }
+            }
+
+            console.log( `::set-output name=changelogsToBeDeleted::${ changelogsToBeDeleted }` )
+
+      - name: Delete changelog files from cherry pick branch
+        run: git rm ${{ steps.changelog.outputs.changelogsToBeDeleted }}
+
+      - name: Commit changes for cherry pick
+        run: git commit --no-verify -am "Prep for cherry pick ${{ needs.prep.outputs.pr }}"
+
+      - name: Push cherry pick branch up
+        run: git push origin cherry-pick-${{ needs.prep.outputs.version }}/${{ needs.prep.outputs.pr }}
+
+      - name: Create the PR for cherry pick branch
+        id: cherry-pick-pr
+        uses: actions/github-script@v6
+        with:
+          script: |
+            let cherryPickPRBody = "This PR cherry-picks the following PRs into the release branch:\n";
+
+            cherryPickPRBody = `${cherryPickPRBody}` + `* #${{ needs.prep.outputs.pr }}` + "\n";
+
+            const pr = await github.rest.pulls.create({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              title: "Prep for cherry pick ${{ needs.prep.outputs.pr }}",
```

Minor nit, can we change the PR title here? The PR doesn't really prep for cherry-picking, it cherry picks. Maybe something like this...?
```suggestion
              title: "Cherry pick ${{ needs.prep.outputs.pr }} into ${{ needs.prep.outputs.release }}",
```